vip-helpers/vip-utils.php

```diff
@@ -718,26 +718,29 @@ function vip_get_random_posts( $number = 1, $post_type = 'post', $return_ids = f
 }
 
 /**
- * This is a sophisticated extended version of wp_remote_get(). It is designed to more gracefully handle failure than wpcom_vip_file_get_contents() does.
+ * This is a sophisticated extended version of wp_remote_request(). It is designed to more gracefully handle failure than wpcom_vip_file_get_contents() does.
  *
- * Note that like wp_remote_get(), this function does not cache.
+ * Note that like wp_remote_request(), this function does not cache.
  *
  * @author tottdev
  * @link http://vip.wordpress.com/documentation/fetching-remote-data/ Fetching Remote Data
- * @param string $url URL to fetch
+ * @param string $url URL to request
  * @param string $fallback_value Optional. Set a fallback value to be returned if the external request fails.
  * @param int $threshold Optional. The number of fails required before subsequent requests automatically return the fallback value. Defaults to 3, with a maximum of 10.
  * @param int $timeout Optional. Number of seconds before the request times out. Valid values 1-3; defaults to 1.
  * @param int $retry Optional. Number of seconds before resetting the fail counter and the number of seconds to delay making new requests after the fail threshold is reached. Defaults to 20, with a minimum of 10.
- * @param array Optional. Set other arguments to be passed to wp_remote_get().
+ * @param array Optional. Set other arguments to be passed to wp_remote_request().
  * @return string|WP_Error|array Array of results. If fail counter is met, returns the $fallback_value, otherwise return WP_Error.
- * @see wp_remote_get()
+ * @see wp_remote_request()
  */
-function vip_safe_wp_remote_get( $url, $fallback_value='', $threshold=3, $timeout=1, $retry=20, $args = array() ) {
+function vip_safe_wp_remote_request( $url, $fallback_value='', $threshold=3, $timeout=1, $retry=20, $args = array() ) {
 	global $blog_id;
 
-	$cache_group = "$blog_id:vip_safe_wp_remote_get";
-	$cache_key = 'disable_remote_get_' . md5( parse_url( $url, PHP_URL_HOST ) );
+	$default_args = array( 'method' => 'GET' );
+	$parsed_args = wp_parse_args( $args, $default_args );
+
+	$cache_group = "$blog_id:vip_safe_wp_remote_request";
```

Comment:
Any concerns around this causing a mass cache invalidation?

Reply:
I'm not too concerned - the function only caches (and for 20 seconds by default) when the remote server is down or responding slowly (which theoretically is only a small portion of requests) - regular requests won't be relying on a cache entry, and it expires quickly anyway.

But, would be fine to leave the group as-is too.